packages/mdc-chips/chip/foundation.ts

```diff
@@ -190,12 +227,104 @@ export class MDCChipFoundation extends MDCFoundation<MDCChipAdapter> {
    */
   handleTrailingIconInteraction(evt: MouseEvent | KeyboardEvent) {
     const isEnter = (evt as KeyboardEvent).key === 'Enter' || (evt as KeyboardEvent).keyCode === 13;
-    evt.stopPropagation();
     if (evt.type === 'click' || isEnter) {
-      this.adapter_.notifyTrailingIconInteraction();
-      if (this.shouldRemoveOnTrailingIconClick_) {
-        this.beginExit();
-      }
+      this.removeChip_(evt);
+    }
+  }
+
+  /**
+   * Handles a keydown event from the root element.
+   */
+  handleKeydown(evt: KeyboardEvent) {
+    const key = evt.key;
+    // Early exit if the key is not usable
+    if (!NAVIGATION_KEYS.has(key)) {
+      return;
+    }
+
+    // Prevent default behavior for movement keys which could include scrolling
+    evt.preventDefault();
+    this.focusNextAction_(key);
+  }
+
+  focusAction(key: string, source: FocusSource) {
+    // Early exit if the key is not usable
+    if (!NAVIGATION_KEYS.has(key)) {
+      return;
+    }
+
+    const shouldJumpChips = JUMP_CHIP_KEYS.has(key);
+    const chipHasTrailingIcon = this.adapter_.hasTrailingIcon();
+    if (shouldJumpChips && (source === FocusSource.Text || !chipHasTrailingIcon)) {
+      return this.focusChipText_();
+    }
+
+    if (shouldJumpChips && source === FocusSource.TrailingIcon && chipHasTrailingIcon) {
+      return this.focusTrailingIcon_();
+    }
+
+    this.focusNextAction_(key);
+  }
+
+  private focusNextAction_(key: string) {
+    const chipIsRTL = this.adapter_.isRTL();
+    const chipHasTrailingIcon = this.adapter_.hasTrailingIcon();
+    const focusSource = this.getFocusSource_();
+    const keyIsLeft = key === strings.ARROW_LEFT_KEY && !chipIsRTL || key === strings.ARROW_RIGHT_KEY && chipIsRTL;
```

Comment:
instead of `keyIsleft` or `keyIsRight`, can you change it to increment/decrement?